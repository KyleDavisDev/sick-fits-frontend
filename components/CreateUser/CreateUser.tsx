import * as React from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Form from "../Form/Form";
import ErrorMessage from "../ErrorMessage/ErrorMessage";

const CREATEUSER_MUTATION = gql`
  mutation CREATEUSER_MUTATION(
    $email: String!
    $name: String!
    $password: String!
  ) {
    createNewUser(email: $email, name: $name, password: $password) {
      id
      email
      name
    }
  }
`;

export interface ICreateUserProps {}

export interface ICreateUserState {
  email: string;
  name: string;
  password: string;
  [name: string]: string;
}

export default class CreateUser extends React.Component<
  ICreateUserProps,
  ICreateUserState
> {
  state = { name: "", password: "", email: "" };

  public render() {
    return (
      <Mutation mutation={CREATEUSER_MUTATION} variables={this.state}>
        {(signup, { error, loading }) => {
          return (
            <Form
              onSubmit={async e => {
                e.preventDefault();
                await signup();
                // reset form
                this.setState({ name: "", email: "", password: "" });
              }}
            >
              <fieldset disabled={loading} aria-busy={loading}>
                <ErrorMessage error={error} />
                <h2>Sign up for an Acount</h2>
                <label htmlFor="email">
                  Email
                  <input
                    type="text"
                    name="email"
                    value={this.state.email}
                    onChange={this.saveToState}
                  ></input>
                </label>

                <label htmlFor="name">
                  Name
                  <input
                    type="text"
                    name="name"
                    value={this.state.name}
                    onChange={this.saveToState}
                  ></input>
                </label>

                <label htmlFor="password">
                  Password
                  <input
                    type="password"
                    name="password"
                    value={this.state.password}
                    onChange={this.saveToState}
                  ></input>
                </label>

                <button type="submit">Submit</button>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }

  private saveToState = (event: React.ChangeEvent<HTMLInputElement>) => {
    // grab name and value
    const name = event.target.name;
    const val = event.target.value;

    // set state
    this.setState({ [name]: val });
  };
}
