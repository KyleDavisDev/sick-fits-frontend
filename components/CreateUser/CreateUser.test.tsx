import "jsdom-global/register";
import * as React from "react";
import { mount } from "enzyme";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import CreateUser, { CREATEUSER_MUTATION } from "./CreateUser";

describe("<CreateUser />", () => {
  it("renders", () => {
    const wrapper = mount(
      <MockedProvider>
        <CreateUser />
      </MockedProvider>
    );

    expect(wrapper).toBeTruthy();
  });

  it("renders", () => {
    const wrapper = mount(
      <MockedProvider>
        <CreateUser />
      </MockedProvider>
    );

    console.log(wrapper.debug());

    expect(wrapper.find("form")).toMatchSnapshot();
  });
});
