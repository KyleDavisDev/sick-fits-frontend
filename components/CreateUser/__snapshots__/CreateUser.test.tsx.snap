// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUser /> renders 1`] = `
<form
  className="sc-bdVaJa cgcGkz"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <ErrorMessage
      error={Object {}}
    />
    <h2>
      Sign up for an Acount
    </h2>
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="name"
    >
      Name
      <input
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </fieldset>
</form>
`;
